sprimber-engine/src/main/java/com/griddynamics/qa/sprimber/reporting/AllureSprimber.java

```diff
@@ -175,39 +175,69 @@ private void doWithTestFinish(Node node) {
+        List<Label> metaLabels = node.getAttribute("meta").map(o -> {
             Node.Meta meta = (Node.Meta) o;
             List<Label> labels = new ArrayList<>();
-            Label epicLabel = ResultsUtils.createEpicLabel(meta.getSingleValueOrDefault("epic", EPIC_DEFAULT_NAME));
-            Label featureLabel = ResultsUtils.createFeatureLabel(
-                    meta.getSingleValueOrDefault("feature", testParentsById.get(node.getParentId().toString()).getParentName()));
-            Label storyLabel = ResultsUtils.createStoryLabel(meta.getSingleValueOrDefault("story", node.getName()));
 
             Label packageLabel = ResultsUtils.createPackageLabel(
                     meta.getSingleValueOrDefault("package", testParentsById.get(node.getParentId().toString()).getParentName()));
-
-            Label parentSuiteLabel = ResultsUtils.createParentSuiteLabel(meta.getSingleValueOrDefault("parentSuite", CUCUMBER_SUITES));
-            Label suiteLabel = ResultsUtils.createSuiteLabel(
-                    meta.getSingleValueOrDefault("suite", testParentsById.get(node.getParentId().toString()).getParentName()));
-            Label subSuiteLabel = ResultsUtils.createSubSuiteLabel(meta.getSingleValueOrDefault("subSuite", node.getName()));
+            labels.add(packageLabel);
 
             Label severityLabel = ResultsUtils.createSeverityLabel(meta.getSingleValueOrDefault("severity", SeverityLevel.NORMAL.value()));
+            labels.add(severityLabel);
 
             Label threadLabel = ResultsUtils.createThreadLabel();
+            labels.add(threadLabel);
             Label hostLabel = ResultsUtils.createHostLabel();
+            labels.add(hostLabel);
+
+            labels.addAll(behaviourPluginLabelList(node, meta));
+            labels.addAll(suitePluginLabelList(node, meta));
 
+            return labels;
+        }).orElse(Collections.emptyList());
+        List<Label> tagLabels = node.getAttribute("bddTags").map(o -> {
+            List<String> tags = (List<String>) o;
+            return tags.stream()
+                    .map(ResultsUtils::createTagLabel)
+                    .collect(Collectors.toList());
+        }).orElse(Collections.emptyList());
+        labelList.addAll(metaLabels);
+        labelList.addAll(tagLabels);
+        return labelList;
+    }
+
+    private List<Label> behaviourPluginLabelList(Node node, Node.Meta meta) {
+        List<Label> labels = new ArrayList<>();
+        if (!meta.getSingleValueOrEmpty("epic").isEmpty()) {
+            Label epicLabel = ResultsUtils.createEpicLabel(meta.getSingleValueOrEmpty("epic"));
             labels.add(epicLabel);
-            labels.add(featureLabel);
+        }
+        Label featureLabel = ResultsUtils.createFeatureLabel(
+                meta.getSingleValueOrDefault("feature", testParentsById.get(node.getParentId().toString()).getParentName()));
+        labels.add(featureLabel);
+        if (!meta.getSingleValueOrEmpty("story").isEmpty()) {
```

Comment:
Please avoid usage of some magic strungs

Reply:
it will gone in the next review